### ISSUE TITLE
Is adding validation checks for json files considered a feature freeze violation?

### ISSUE THREAD

**Issue by logical-1985516** [nus-cs2103-AY2324S2/forum#842]
Hi Prof, may I ask whether it is ok to add code to perform checks on the json objects, so that certain edits to the data file would not cause the app to crash? E.g. checking if a certain attribute in a json object is null to prevent potential NPEs. Adding on to #837.

**Comment by damithc** [comment]
@logical-1985516 Yes, it would be enhancing a feature, as the crash prevention is an enhancement that is currently neither implemented nor promised in the UG.

**Comment by logical-1985516** [comment]
Thank you Prof for your response. Would this be considered a bug/feature flaw?

**Comment by damithc** [comment]
_Enhancement_ implies feature flaw.
